wami/app.js

```diff
@@ -414,6 +432,208 @@ viewImagesButton.addEventListener('click', async e => {
   }
 });
 
+// Process share target data when the app is launched
+async function processShareTargetData() {
+  // Check if we were launched via share target
+  const urlParams = new URLSearchParams(window.location.search);
+  const isShared = urlParams.get('share') === 'true';
+  
+  if (isShared) {
+    try {
+      // Get the shared data from the cache
+      const shareCache = await caches.open('share-target-cache');
+      const shareDataResponse = await shareCache.match('shareData');
+      
+      if (shareDataResponse) {
+        const shareData = await shareDataResponse.json();
+        console.log('Share data received:', shareData);
+        
+        if (shareData.fileCount > 0) {
+          // Determine the flow name - check if URL is a web+wami URL
+          let flowTitle = shareData.title || 'Shared Images Flow';
+          
+          // Default flow steps
+          let flowSteps = [
+            {
+              type: 'resize-width-if-larger',
+              params: [1000]
+            }
+          ];
+          
+          // If URL field exists and starts with web+wami://, use it as the flow name
+          if (shareData.url && shareData.url.trim() !== '') {
```

Comment:
The code in this function is pretty long and a bit hard to read. Can you maybe extract some of its subparts into other functions, so that the main flow is easier to read?